runtime/sync_win32.h

```diff
@@ -45,23 +46,39 @@ Caml_inline int sync_mutex_destroy(sync_mutex m)
 
 Caml_inline int sync_mutex_lock(sync_mutex m)
 {
-  AcquireSRWLockExclusive(m);
-  return 0;
+  DWORD self_tid = GetCurrentThreadId();
```

Comment:
The use of `GetCurrentThreadId` resurrects the conversation started in https://github.com/ocaml/ocaml/pull/9846#pullrequestreview-484124107 - there were two concerns. Firstly was whether it was theoretically possible to attack the thread ID and secondly was whether `GetCurrentThreadId` (accessing the TCB and so presumably(?) having to cross into kernel mode) is slower than TLS (which doesn't).

The difference is that we don't have thread descriptors necessarily here - but we could simply have a TLS value in `platform.c` and ensure that the thread creation function sets it? (in this case, the thread creation function obviously mustn't be inlined)